ra/ra.go

```diff
@@ -1079,6 +1080,11 @@ func (ra *RegistrationAuthorityImpl) onValidationUpdate(ctx context.Context, aut
 	return nil
 }
 
+// DeactivateRegistration deactivates a valid registration
+func (ra *RegistrationAuthorityImpl) DeactivateRegistration(ctx context.Context, id int64) error {
+	return ra.SA.DeactivateRegistration(ctx, id)
```

Comment:
I notice that for `DeactivateAuthorization` you pass the `auth` object to the RA and check the status is `core.StatusValid` or `core.StatusPending`, returning a `MalformedRequestError` if the `auth` has the wrong status.

For `DeactivateRegistration` only the `id` of the reg object is given to the RA and subsequently there is no check on the current `reg` status at the RA level. Instead the WFE checks that the `reg` is `core.StatusValid` before giving it to the RA.

Was there a reason you favoured passing just the ID for the reg deactivation? Right now I'm leaning towards suggesting we keep it consistent and pass a `reg` instead of the ID. Then the RA can enforce the correct status pre-deactivation in addition to the WFE.